### ISSUE TITLE
gdal2tiles.py not support input raster from source other than file

### ISSUE THREAD

**Issue by fechen123** [OSGeo/gdal#6731]
## Expected behavior and actual behavior.

The following is not working now with gdal2tiles.py:

gdal2tiles.py --zoom=2-5 /vsizip/tgr.zip/tgrgtf1.tif tiles
gdal2tiles.py --zoom=2-5 geor:user/pwd,,grtab,grobj,id=0 tiles

gdal2tiles.py: error: The provided input file /vsizip/tgr.zip/tgrgtf1.tif does not exist or is not a file

Not sure it is a bug or it is designed this way. But other python scripts such as gdal_polygonize.py support all the data sources. 

## Steps to reproduce the problem.

- create a zipfile with a tif file in it.
- run the above command. 

## Operating system

Linux

## GDAL version and provenance

This is tested on GDAL3.4.1

**Comment by mdsumner** [comment]
was about to report this with

```
gdal2tiles "vrt://gcore/data/float64.tif?ot=Byte&scale=true"  /vsimem/float

#gdal2tiles.py: error: The provided input file vrt://gcore/data/float64.tif?ot=Byte&scale=true does not exist or is not a file
```

GDAL 3.9.0dev-735e122955, released 2024/04/22

**Comment by rouault** [comment]
has been fixed per 5c0b8d79ba47b7bb2c8e10ed2558a5e8bfeab89b . duplicate of #9808
